### DISCUSSION TITLE
MapReduce Benchmark : object key does not exist

### DISCUSSION THREAD

**Discussion by jingren1021** [vhive-serverless/vSwarm#351] [Q&A] [answered]
As my access to "storage-module-test" S3 bucket in MapReduce benchmark was denied, I modified the code to use custom S3 bucket.
After running locally, I came across error as below.

```
jingren@hp-worker:~/vSwarm$ kubectl logs -n default -c user-container -l serving.knative.dev/service=mapper --tail=-1
INFO:root:Using inline or s3 transfers
INFO:root:Mapper 6 is invoked
[key: "part-00006"
] 6
ERROR:grpc._server:Exception calling application: An error occurred (NoSuchKey) when calling the GetObject operation: The specified key does not exist.
```

Is there any initial object that should be inserted into custom S3 bucket before running MapReduce benchmark or is it some other error?

**Comment by ustiugov** [comment, marked as answer]
@jingren1021 It uses one of the datasets from AMPLab Big Data Bench: https://amplab.cs.berkeley.edu/benchmark . If I recall correctly, we used `1node` dataset.

**Reply by jingren1021** [reply]
The bucket used to store AMPLab datasets appear to be private now. I noticed that an S3 bucket named "storage-module-test" is used in vSwarm MapReduce benchmark. If the "storage-module-test" is managed by vhive-serverless team, is it possible to allow getObject operation temporarily for users to port dataset to their own Amazon S3 buckets.

**Reply by ustiugov** [reply]
@dhschall can you please help with that?

**Reply by dhschall** [reply]
@shyamjesal you worked on the storage module.
Do you know if this is possible?

**Reply by ustiugov** [reply]
@shyamjesal

**Reply by shyamjesal** [reply]
As far as I remember, we do need some data initially in the bucket. We can allow get access to that.

**Reply by ustiugov** [reply]
@jingren1021 please DM Shyam on our slack if you still need it.

**Reply by jingren1021** [reply]
I don't need the dataset anymore. Thanks for the help!

**Reply by shyamjesal** [reply]
Hi @jingren1021, we sadly lost the access the the dataset. Did you manage to download it?

**Reply by jingren1021** [reply]
I didn't manage to download the dataset.

**Reply by ustiugov** [reply]
@jingren1021 soon after we did manage to find a backup. let us know if you need it :)
